### DISCUSSION TITLE
Ability to assign custom class(es) to fields for enhanced theming capabilities

### DISCUSSION THREAD

**Discussion by formfcw** [directus/directus#22321] [Feature Requests]
### Summary

This is a small improvement. As the title describes it, it would be useful to be able to assign class(es) to fields, so that we can style them individually for the item page.

### Motivation

While working with Directus, I often found that I needed to change the default appearance of fields on the item page. So I wrote the [Classified Group Interface](https://github.com/formfcw/directus-extension-classified-group) extension. The extension is a group interface that renders custom classes to that group. This way we can apply custom CSS to fields, that are wrapped by the group.

### Detailed Design

👉  When creating a field, users can add custom classes from the field tab.

![field settings](https://github.com/directus/directus/assets/78852214/87d4b988-80ac-483c-80db-fa4dee90b1ed)

👉 In the `Settings > Appearance > Custom CSS` field users can add the appropriate CSS.

![appearance](https://github.com/directus/directus/assets/78852214/d72ed358-a3b7-45e6-8ccc-e850ab09df4d)

👉 On the item page, the field appears with the custom CSS, because the `.field` div also contains the custom classes.

![item view](https://github.com/directus/directus/assets/78852214/818d6de3-db05-4583-921e-7cd13b11e016)

### Must Have:
- A field in the field settings where users can add custom classes

### Drawbacks

I think this is a small improvement that could be very useful. The only drawback I can think of is that applied classes could somehow affect future changes: if a user uses a class that Directus will implement in the future. But as long as it is scoped, this won't be a problem.

### Alternatives

Maybe the `field` tab is not the perfect place, because it should only affect fields on the item page – so interfaces and not displays.

### Adoption Strategy

Won’t break anything

### Unresolved Questions

Is the `field` tab in the appearance settings the perfect place for the input field? (see above)

**Comment by Abdallah-Awwad** [comment]
Great idea! 
In addition to adding custom CSS to the field `value`, I'd also like to apply custom CSS to the field `key` so they look more user-friendly together.

**Comment by wolfgang-hartl** [comment]
Another use case for me is that with the help of the extension of formfcw I was able to hide fields in an m-2-m relationship. I simply grouped them, assigned a CSS class to this group, and then hid or disabled them with the following CSS rules. 

```
#dialog-outlet .hidden-on-relation{
    display: none;
}
#dialog-outlet .disabled-on-relation{
    pointer-events: none;
}
```

Super helpful when I only want to display the junction fields and a few additional fields that should not be editable!

![image](https://github.com/directus/directus/assets/75800215/9f74947f-a965-41e3-aa8f-a2742a55478f)

![image](https://github.com/directus/directus/assets/75800215/1262fc3d-96c7-48d6-bda9-42d20d953e18)

**Comment by JamesW1** [comment]
Closing this discussion as this has now been resolved. Directus now includes data-primary-key, data-collection, and data-field attributes on field elements, allowing programmatic identification and interaction with fields in the UI.

Example:
<img width="3222" height="778" alt="image" src="https://github.com/user-attachments/assets/456b396d-c95b-47af-a77d-b18f0870dce1" />
